FREnctools_lib/pyfrenctools/make_hgrid/make_hgrid_wrappers.py

```diff
@@ -0,0 +1,877 @@
+import numpy as np
+import numpy.typing as npt
+import ctypes
+
+from pyfms.pyfms_utils.data_handling import (
+    setscalar_Cint32,
+    setscalar_Cdouble,
+    setarray_Cdouble,
+    setarray_Cint32,
+    set_Cchar,
+)
+
+# lib = ctypes.CDLL("./FREnctools_lib/cfrenctools/c_build/clib.so")
+
+class make_hgrid_wrappers():
+    __libpath: str = None
+    __lib: ctypes.CDLL = None
+
+    @classmethod
+    def init(cls, libpath, lib):
+        cls.__libpath = libpath
+        cls.__lib = lib
+
+    @classmethod
+    @property
+    def lib(cls):
+        return cls.__lib
+
+    @classmethod
+    @property
+    def libpath(cls):
+        return cls.__libpath
+    
+    @classmethod
+    def fill_cubic_grid_halo(
+            cls,
+            nx: int, 
+            ny: int, 
+            halo: int, 
+            data: npt.NDArray[np.float64], 
+            data1_all: npt.NDArray[np.float64],
+            data2_all: npt.NDArray[np.float64],
+            tile: int,
+            ioff: int,
+            joff: int,
+    ):
+        nxp = nx + ioff
+        nyp = ny + joff
+        nxph = nx + ioff + 2*halo
+        nyph = ny + joff + 2*halo
+        
+        for i in range(nxph*nyph):
+            data[i] = -9999.
+
+        # first copy computing domain data
+        for j in range (nyp+1):
+            for i in range(nxp+1):
+                data[j*nxph+i] = data1_all[tile*nxp*nyp+(j-1)*nxp+(i-1)]
+                
+        ntiles = 6
+        
+        if tile%2 == 1:
+            lw = (tile+ntiles-1)%ntiles
+            le = (tile+ntiles+2)%ntiles
+            ls = (tile+ntiles-2)%ntiles
+            ln = (tile+ntiles+1)%ntiles
+            for j in range(nyp+1):
+                data[j*nxph] = data1_all[lw*nxp*nyp+(j-1)*nxp+nx-1]       # west halo
+                data[j*nxph+nxp+1] = data2_all[le*nxp*nyp+ioff*nxp+nyp-j] # east halo
+            for i in range(nxp+1):
+                data[i] = data2_all[ls*nxp*nyp+(nxp-i)*nyp+(nx-1)]        # south halo
+                data[(nyp+1)*nxph+i] = data1_all[ln*nxp*nyp+joff*nxp+i-1] # north halo
+        else:
+            lw = (tile+ntiles-2)%ntiles
+            le = (tile+ntiles+1)%ntiles
+            ls = (tile+ntiles-1)%ntiles
+            ln = (tile+ntiles+2)%ntiles
+            for j in range(nyp+1):
+                data[j*nxph] = data2_all[lw*nxp*nyp+(ny-1)*nxp+nyp-j]     # west halo
+                data[j*nxph+nxp+1] = data1_all[le*nxp*nyp+(j-1)*nxp+ioff] # east halo
+            for i in range(nxp+1):
+                data[i] = data1_all[ls*nxp*nyp+(ny-1)*nxp+i-1]                # south halo
+                data[(nyp+1)*nxph+i] = data2_all[ln*nxp*nyp+(nxp-i)*nyp+joff] # north halo
+
+    @classmethod
+    def create_regular_lonlat_grid(
```

Comment:
could each of these methods be put in their own files? 
For example, the function `create_regular_lonlat_grid` would be in its `lonlat.py` file and the function can be renamed to `create` so that users can do `pyfrenctools.make_hgrid.lonlat.create(...)`